x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/repository/CcrRepository.java

```diff
@@ -389,5 +402,17 @@ public int read(byte[] bytes, int off, int len) throws IOException {
 
             return bytesReceived;
         }
+
+        private void maybePause(int bytesRequested) {
```

Comment:
just to make sure, this is the same code as in PeerRecoveryTargetService. Let's have a follow-up to refactor it into a class that is reusable, and use it everywhere (peer recovery source, target as well as snapshot + restore).

Reply:
I did have to extract a `CombinedRateLimiter`. Otherwise the different repositories for different remote clusters will not share the rate limiter. However, I added a meta issue to ensure that we use the same code base for this and peer recovery. So we can work out the issues in that follow-up?

Reply:
ok, sounds good.